core-web/libs/sdk/client/src/lib/client/content/builders/collection/collection.ts

```diff
@@ -0,0 +1,334 @@
+     * @return {CollectionBuilder} CollectionBuilder - A CollectionBuilder instance
+     * @memberof CollectionBuilder
+     */
+    render(render: boolean): this {
+        this.#render = render;
+
+        return this;
+    }
+
+    /**
+     * Takes an array of constrains to sort the content by field an specific order
+     *
+     * @example
+     * ```javascript
+     * // This will sort the content by title in ascending order
+     * // and by modDate in descending order
+     *  const sortBy = [{ field: 'title', order: 'asc' }, { field: 'modDate', order: 'desc' }]
+     *
+     *  client.content.getCollection("Blog").sortBy(sortBy)
+     *```
+     *
+     * @param {SortBy[]} sortBy Array of constrains to sort the content by
+     * @return {CollectionBuilder} CollectionBuilder - A CollectionBuilder instance
+     * @memberof CollectionBuilder
+     */
+    sortBy(sortBy: SortBy[]): this {
+        this.#sortBy = sortBy;
+
+        return this;
+    }
+
+    /**
+     * Takes a number that represents the max amount of content to fetch
+     *
+     * Note: The limit is set to 10 by default
+     *
+     * @param {number} limit The max amount of content to fetch
+     * @return {CollectionBuilder} CollectionBuilder - A CollectionBuilder instance
+     * @memberof CollectionBuilder
+     */
+    limit(limit: number): this {
+        this.#limit = limit;
+
+        return this;
+    }
+
+    /**
+     * Takes a number that represents the page to fetch
+     *
+     * @param {number} page The page to fetch
+     * @return {CollectionBuilder} CollectionBuilder - A CollectionBuilder instance
+     * @memberof CollectionBuilder
+     */
+    page(page: number): this {
+        this.#page = page;
+
+        return this;
+    }
+
+    /**
+     * Takes a string that represents a Lucene Query that is used to filter the content to fetch.
+     *
+     * Note: The string is not validated, so be cautious when using it.
+     *
+     * @param {string} query A Lucene Query String
+     * @return {CollectionBuilder} CollectionBuilder - A CollectionBuilder instance
+     * @memberof CollectionBuilder
+     */
+    query(query: string): this;
+
+    /**
+     * Takes a function that recieves a QueryBuilder to buid a query for content filtering.
+     * @example
+     *```javascript
+     * // This will filter the content by title equals 'Hello World' or 'Hello World 2'
+     * client.content.getCollection("Activity").query((queryBuilder) =>
+     *     queryBuilder.field('title').equals('Hello World').or().equals('Hello World 2')
+     * );
+     *```
+     * @param {BuildQuery} buildQuery A function that receives a QueryBuilder instance and returns a valid query
+     * @return {CollectionBuilder} CollectionBuilder - A CollectionBuilder instance
+     * @memberof CollectionBuilder
+     */
+    query(buildQuery: BuildQuery): this;
+    query(buildQuery: BuildQuery | string): this {
```

Comment:
```suggestion
    query(buildQuery: unknown): this {
```

No expert in overloading... but are we sure about this? because if we add another overload we will have to come and update params here.

Reply:
If we can pass a string for a raw lucene query, why do we need that `BuildQuery` have a `raw` method?

Reply:
`query` with string params is totally different and isolated from the query method with a function until the end when they get mixed together. If you want to append a complex query to the builder, you will need the `raw` method

Reply:
The overloading is fine, I understand what we're doing here... but in the actual implementation signature you should be using `unknown` and not repeat the types, because the overloading is what do that.

Reply:
Yep I changed it to `unknown`